src/updater.js

```diff
@@ -0,0 +1,55 @@
+const { dialog } = require('electron');
+const { autoUpdater } = require('electron-updater');
+const noUpdateDialog = require('./no_update_dialog.js');
+const updateAvailableDialog = require('./update_available_dialog.js');
+
+// Pointer to menuItem so we can manage it's state.
+let updateMenuItem = null;
+// If menu is clicked on we'll set the event. Used to decide about showing dialog.
+let updateEvent = null;
+
+let updateInfo = null;
+
+autoUpdater.autoDownload = false;
+
+autoUpdater.on('error', (error) => {
+  // TODO: This needs ux for a better user experience.
+  dialog.showErrorBox('Error: ', error == null ? "unknown" : (error.stack || error).toString());
+  // Re-enable menuItem since something went wrong while checking.
+  updateMenuItem.enabled = true;
+})
+
+autoUpdater.on('update-available', (updateInfo) => {
+  updateInfo = updateInfo;
```

Comment:
Is this needed?

Reply:
I had run ahead a little bit and was using it I think it can be pulled out of this one.